packages/plugin-ext/src/plugin/tasks/tasks.ts

```diff
@@ -25,22 +25,19 @@ import * as theia from '@theia/plugin';
 import * as converter from '../type-converters';
 import { CustomExecution, Disposable } from '../types-impl';
 import { RPCProtocol, ConnectionClosedError } from '../../common/rpc-protocol';
-import { TaskProviderAdapter } from './task-provider';
 import { Emitter, Event } from '@theia/core/lib/common/event';
 import { TerminalServiceExtImpl } from '../terminal-ext';
 import { UUID } from '@theia/core/shared/@phosphor/coreutils';
-
-type ExecutionCallback = (resolvedDefinition: theia.TaskDefinition) => Thenable<theia.Pseudoterminal>;
 export class TasksExtImpl implements TasksExt {
     private proxy: TasksMain;
 
     private callId = 0;
-    private adaptersMap = new Map<number, TaskProviderAdapter>();
+    private providersByHandle = new Map<number, theia.TaskProvider>();
     private executions = new Map<number, theia.TaskExecution>();
     protected callbackIdBase: string = UUID.uuid4();
-    protected callbackId: number;
-    protected customExecutionIds: Map<ExecutionCallback, string> = new Map();
-    protected customExecutionFunctions: Map<string, ExecutionCallback> = new Map();
+    protected callbackId: number = 0;
+    protected providedCustomExecutions: Map<string, theia.CustomExecution> = new Map();
+    protected oneOffCustomExecutions: Map<string, theia.CustomExecution> = new Map();
```

Comment:
`oneOffCustomExecutions`-  is actually a storage for custom executions of tasks which **are running** - correct?

Reply:
No. On-off executions are those provided through "executeTask". Clients may construct a new custom execution that is not related to the ones provided through "resolveTask" or "provideTasks".